jupyter/jupyter.go

```diff
@@ -299,6 +296,27 @@ func (s *server) execute(c *conn, req *message) error {
 		return s.publishIO("error", content, req)
 	}
 
+	r, w, err := os.Pipe()
+	if err != nil {
+		return errResp(err)
+	}
+
+	session.Stdout = w
+	session.Stderr = w
+	stdout := os.Stdout
+	os.Stdout = w
+
+	err = session.Exec([]byte(reqContent["code"].(string)))
+
+	w.Close()
+	buf := new(bytes.Buffer)
+	io.Copy(buf, r)
```

Comment:
I believe this can get stuck.

Try executing a neugram program via jupyter that writes more than 64kb to stdout. It will never get out of session.Exec, because the linux kernel buffer for the pipe will fill up, and it will block waiting for something to drain it. You don't drain it until here, which will never be reached.

I think you need to start a goroutine for the io.Copy before you call session.Exec, and then use a channel close trick to wait for it here.